### PR TITLE
Issue #171 - Fixing Some Lint Warnings

### PR DESCRIPTION
**Pull request by JasonSarwar**
**Description (required)**

Fixes part of #171 Fix Lint errors/warnings

Fixing Some Android Lint and SonarLint Warnings in these packages:
* `fr.free.nrw.commons.bookmarks`
* `fr.free.nrw.commons.bookmarks.locations`
* `fr.free.nrw.commons.bookmarks.pictures`
* `fr.free.nrw.commons.category`

Some of the changes were:
* Switching the try/catch/finally blocks for database querying into try-with-resources blocks, with the `Cursor` object getting set as the closable resource
* Switching a lot of field types in the sqlite "Create Table Statements" from STRING to TEXT
* Creating `DAOException` and throwing that instead of a generic `RuntimeException` in a few places ([SonarLint: RSPEC-112](https://rules.sonarsource.com/java/RSPEC-112))
* Logging some Exceptions using Timber instead of `e.printStackTrace()` ([SonarLint: RSPEC-1148](https://rules.sonarsource.com/java/RSPEC-1148))
* Changing a few access modifiers to be more restrictive
* Adding `final` to a few variables that are never changed
* Adding a private constructor to some utility or static inner classes ([SonarLint: RSPEC-1118](https://rules.sonarsource.com/java/RSPEC-1118)) 
* Made some constant String variables for recurring Strings ([SonarLint: RSPEC-1192](https://rules.sonarsource.com/java/RSPEC-1192))
* Changing the switch cases with only one case into if/else statements ([SonarLint: RSPEC-1301](https://rules.sonarsource.com/java/RSPEC-1301))
* Adding some `@NonNull` and `@Nullable` annotations to methods that inherit methods that use those annotations
* Replacing some `junit.framework` package imports in some Kotlin test classes with `org.junit` because the former is deprecated

**Tests performed (required)**
Ran the Gradle Tests and Tested ProdDebug on the Nexus 6 Emulator with API level 22.

---

Please let me know if there's anything I should change back or if there's anything I should elaborate on!

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/category/CategoriesModel.java [hunk 1]
```diff
@@ -122,7 +123,7 @@ boolean cacheContainsKey(String term) {
             Observable<CategoryItem> categoryItemObservable = gpsCategories()
                     .concatWith(titleCategories(imageTitleList));
             if (hasDirectCategories()) {
-                categoryItemObservable.concatWith(directCategories().concatWith(recentCategories()));
+                categoryItemObservable = categoryItemObservable.concatWith(directCategories().concatWith(recentCategories()));
```
Comment (JasonSarwar) [inline app/src/main/java/fr/free/nrw/commons/category/CategoriesModel.java]: This one's a bit of logic change. Just wanted to point this one out.

**Comment by JasonSarwar** [comment]
Fixing conflicts now.

**Comment by madhurgupta10** [comment]
@JasonSarwar Let us know if you are still working on it!

**Comment by madhurgupta10** [comment]
I am closing the PR for now as it seems like the author isn't working on it anymore, feel free to reopen if needed :)

**Pull request closed without merging**